File: Code/src/domain/entities.py
from dataclasses import dataclass
from pathlib import Path


@dataclass(frozen=True)
class PredictionItem:
    label: str
    confidence: float


@dataclass(frozen=True)
class PredictionResult:
    image_path: Path
    predictions: list[PredictionItem]

    @property
    def top_prediction(self) -> PredictionItem:
        return self.predictions[0]

File: App/lib/domain/Entities.dart
class PredictionItem {
  final String label;
  final double confidence;

  const PredictionItem({required this.label, required this.confidence});
}

class PredictionResult {
  final List<PredictionItem> predictions;
  final String imagePath;

  const PredictionResult({required this.predictions, required this.imagePath});

  PredictionItem get topPrediction => predictions.isNotEmpty
      ? predictions.first
      : const PredictionItem(label: '', confidence: 0);
}
